internal/fsutil/path.go

```diff
@@ -0,0 +1,47 @@
+package fsutil
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// These functions exists because we work with slash terminated directory paths
+// that come from deb package tarballs but standard library path functions
+// treat slash terminated paths as unclean.
+
+// Like filepath.Dir() but trailing slash on input is ignored and the return
+// value always includes trailing slash.
+//
+// Comparison of filepath.Dir() with fsutil.Dir():
+//
+//	filepath.Dir("/foo/bar/")	== "/foo/bar"
+//	filepath.Dir("/foo/bar")	== "/foo"
+//
+//	fsutil.Dir("/foo/bar")		== "/foo/"
+//	fsutil.Dir("/foo/bar/")		== "/foo/"
```

Comment:
Again, this comment makes sense in the PR description, but not really in the code. An example of what this `Dir` function returns would be good, but I don't personally see much reason to have the comparison. When someone wants to use this function they just need to know what its going to return, its reason for existing should be obvious from that (i.e. they can look at the code and see that `filepath.Dir` is called inside the function).

Reply:
Replied above. Resolving this one.